### PR TITLE
MGDAPI-4602 Fixed S1028 linter errors

### PR DESCRIPTION
**Pull request by carlkyrillos**
# Issue link
JIRA: https://issues.redhat.com/browse/MGDAPI-4602

# What
This PR addresses the S1028 errors that were previously configured to be ignored by our linter. This PR also updates our linter [config file](https://github.com/integr8ly/integreatly-operator/blob/master/.golangci.yaml#L11) to check for S1028 errors going forward.

# Verification steps

- An eye review of the changes and passing the PROW checks should be sufficient but if you want to manually verify the changes:
  - Checkout this branch
  - Run `make test/lint`
  - Confirm there are no errors in the output

### CODE REVIEW

File: pkg/products/rhssocommon/reconciler_test.go [hunk 1]
```diff
@@ -1114,18 +1114,18 @@ func TestReconciler_ReconcileCSVEnvVars(t *testing.T) {
...
+							return fmt.Errorf("expected env var {PAULS_AMAZEBALLS_ENV_VAR: yeaaah}, got %+v", envs)
 						}
 					} else {
-						return errors.New(fmt.Sprintf("unexpected env var (should only have PHILS_COOL_ENV_VAR and PAULS_AMAZEBALLS_ENV_VAR): %+v", envs))
+						return fmt.Errorf("unexpected env var (should only have PHILS_COOL_ENV_VAR and PAULS_AMAZEBALLS_ENV_VAR): %+v", envs)
 					}
 				}
 				if !updated {
-					return errors.New(fmt.Sprintf("Expected updated to be true, but got: %v", updated))
+					return fmt.Errorf("wxpected updated to be true, but got: %v", updated)
```
Comment (Fiona-Waters) [inline pkg/products/rhssocommon/reconciler_test.go:1128]:
```suggestion
					return fmt.Errorf("expected updated to be true, but got: %v", updated)
```

Comment (Fiona-Waters) [reply pkg/products/rhssocommon/reconciler_test.go:1128]: Running make test/lint returns no errors. Happy to add lgtm when above is updated.

Comment (carlkyrillos) [reply pkg/products/rhssocommon/reconciler_test.go]: Nice catch thank you! I just pushed the fix

**Comment by Fiona-Waters** [comment]
Running make test/lint returns no errors. Code changes look good.
/lgtm

**Pull request merged**